### DISCUSSION TITLE
Why Github Pages Doesn't Support Django?

### DISCUSSION THREAD

**Discussion by Amir-Mirfallahi** [community/community#54081] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I think if github add the new feature to use python and Django on github pages is very good for developers

**Comment by anasrawuthar** [comment]
i have also same question

**Reply by Amir-Mirfallahi** [reply]
if you wan't to test your project you can also use pythonanywhere. but I found a new one that's very cool
that's runflare. (it's an iranian host which support a lot of framework such as Django, laravell, etc) it also support English if you want.
the deployment also is very easy. you just run `pip install runflare` and after that you go to ur main directory and you run `runflare deploy`.
it's the most easy way to test project. also it has premium account but at first you will get 500,000 Rial to test your project. (after the trial end you will need to charge your account.) if you had any problem with buying premium I can help you. 
good luck :)

**Reply by GhostOf0days** [reply]
Don't use non-industry standard deployment solutions. Use Netlify, Heroku, or Vercel.

**Reply by Amir-Mirfallahi** [reply]
They're God in deployment

**Comment by GhostOf0days** [comment]
Github Pages is very limited. Please use Netlify, Heroku, or Vercel instead. Use Netlify for small, simple use cases/websites/web apps and Vercel for large, complex use cases/websites/web apps.

**Comment by AndyPendragon** [comment]
**Github pages** is a _static_ site hosting service that only supports static HTML, CSS, and JavaScript files straight from a repository on GitHub.
 **Django**, is _dynamic_ on the other hand, is a Python web framework that is used for building dynamic web applications.

Since Github Pages doesn't support server-side languages like Python, it is not suitable for hosting Django applications

### Popular options for hosting Django is:

- **[Heroku](https://devcenter.heroku.com/articles/deploying-python)**: Heroku is a cloud platform that supports Python and provides easy deployment for Django applications. It offers a free tier for small projects and scales well for larger applications.
- **[PythonAnywhere](https://help.pythonanywhere.com/pages/DeployExistingDjangoProject/)**: PythonAnywhere is a platform specifically designed for hosting Python web applications. It provides a web-based development environment and supports Django deployment.
- **[AWS Elastic Beanstalk](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/create-deploy-python-django.html)**: AWS Elastic Beanstalk is a fully managed service that makes it easy to deploy and run applications in multiple languages, including Python and Django. It provides a scalable and reliable infrastructure for hosting Django applications.

**Reply by GhostOf0days** [reply]
I would say to stick with Netlify, Heroku, or Vercel.
